### PR TITLE
[stdlib][move-prover] Partial specs for LibraAccount + prover fixes

### PR DESCRIPTION
**Pull request by wrwg**
This provides some local specs for LibraAccount and turns verification on for the whole module (before this PR, it did not terminate). The fix for the termination problem is a smarter way to translate byte blob constants.

The local specification provided is for deposit, which is one of the more difficult ones. It required changes in FixedPoint32 and to add spec helper functions at various places.

This also removes assert_is_genesis and replaces it with assert(is_genesis(), 0).

More work on LibraAccount in follow up PR.

## Motivation

Specs

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Existing tests

## Related PRs

NA

### CODE REVIEW

File: language/stdlib/modules/LibraAccount.move [hunk 1]
```diff
@@ -212,17 +217,7 @@ module LibraAccount {
...
-        let approx_lbr_microlibra_value = Libra::approx_lbr_for_value<Token>(deposit_value);
-        let above_threshold = approx_lbr_microlibra_value >= travel_rule_limit_microlibra;
-        // travel rule only applies if the sender and recipient are both VASPs
-        let both_vasps = VASP::is_vasp(payer) && VASP::is_vasp(payee);
-        if (above_threshold &&
-            both_vasps &&
-            // travel rule does not apply for intra-VASP transactions
-            VASP::parent_address(payer) != VASP::parent_address(payee)
-        ) {
+        if (travel_rule_applies<Token>(payer, payee, deposit_value)) {
```
Comment (sblackshear) [inline language/stdlib/modules/LibraAccount.move:220]: Nit: can we call this `dual_attestation_applies`? Although the old comments only mentioned the travel rule, the dual attestation protocol is/will be used for both the travel rule and for other purposes.

File: language/stdlib/modules/RegisteredCurrencies.move [hunk 2]
```diff
@@ -26,7 +26,7 @@ module RegisteredCurrencies {
         config_account: &signer,
         create_config_capability: &Capability<CreateOnChainConfig>,
     ): RegistrationCapability {
-        assert_is_genesis();
+        assert(is_genesis(), 0);
```
Comment (sblackshear) [inline language/stdlib/modules/RegisteredCurrencies.move:29]:
```suggestion
        assert(LibraTimestamp::is_genesis(), 0);
```

File: language/stdlib/modules/LibraAccount.move [hunk 3]
```diff
@@ -267,6 +263,61 @@ module LibraAccount {
...
+        metadata_signature: vector<u8>;
+        aborts_if !spec_has_account_operations_cap();
+        aborts_if to_deposit.value == 0;
+        aborts_if !exists<LibraAccount>(payer);
+        aborts_if !exists<LibraAccount>(payee);
+        aborts_if !exists<Balance<Token>>(payee);
+        aborts_if global<Balance<Token>>(payee).coin.value + to_deposit.value > max_u64();
+        include TravelRuleAppliesAbortsIf<Token>;
+        aborts_if spec_travel_rule_applies<Token>(payer, payee, to_deposit.value)
+                        && len(metadata_signature) != 64;
```
Comment (sblackshear) [inline language/stdlib/modules/LibraAccount.move:283]: Might make sense to extract this to a `spec fun signature_correct(metadata, metadata_signature, payee, amount)` that (for now) just checks the length of `metadata_signature`, but documents the abort conditions that are missing (cryptographic signature verification fails or `metadata != payer | amount | domain_separator`)

**Comment by wrwg** [comment]
This has quite a bit changed, put I plan to go ahead with Sam's approval  in order to not accumulate more merge conflicts.

File: language/stdlib/modules/RegisteredCurrencies.move (hunk 2 above)
Comment (sblackshear) [reply language/stdlib/modules/RegisteredCurrencies.move:27]: Correct!

**Pull request closed without merging**